amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestConsumer.java

```diff
@@ -220,4 +258,53 @@ private void deleteResources(StreamExistenceManager streamExistenceManager, Leas
         log.info("-------------Finished deleting resources.----------------");
     }
 
+    @Data
+    @ToString
+    @AllArgsConstructor
+    public static class StreamScaler implements Runnable {
+        private final KinesisAsyncClient client;
+        private final String streamName;
+        private final Queue<ReshardOptions> scalingFactor;
+        private int totalRotations;
+        private KCLAppConfig testConfig;
+
+        @Override
+        public void run() {
+            try {
+                log.info("----------------------------Starting stream scale----------------------");
+                if (!scalingFactor.isEmpty()) {
+                    log.info("Starting stream scaling with params : {}", this);
+                    final DescribeStreamSummaryRequest describeStreamSummaryRequest = DescribeStreamSummaryRequest.builder().streamName(streamName).build();
+                    final DescribeStreamSummaryResponse response = client.describeStreamSummary(describeStreamSummaryRequest).get();
+
+                    int openShardCount = response.streamDescriptionSummary().openShardCount();
+                    int targetShardCount;
+                    if (scalingFactor.peek() == ReshardOptions.SPLIT) {
+                        // Split case: double the number of shards
+                        targetShardCount = (int) (openShardCount * 2.0);
+                    } else {
+                        // Merge case: half the number of shards
+                        targetShardCount = (int) (openShardCount * 0.5);
+                    }
```

Comment:
Use a smart enum. This avoids distributing, and potentially duplicating, logic all over the codebase. Example:
```
enum ReshardOptions {
   SPLIT {
      int calculateShardCount(int currentShards) {
         return 2 * currentShards;
      }
   },
   ...

   abstract int calculateShardCount(int currentShards);
}
```

This code then simplifies to:
```
final int targetShardCount = scalingFactor.peek().calculateShardCount(openShardCount);
```